lua/nvim-tree.lua

```diff
@@ -511,7 +511,11 @@ local DEFAULT_OPTS = { -- BEGIN_DEFAULT_OPTS
       watcher = false,
     },
   },
-} -- END_DEFAULT_OPTS
+  marks = {
+    enable_persistence = false,
+    save_path = nil, -- nil will default to stdpath("data") .. "/nvim-tree-bookmarks.json"
+  },
```

Comment:
```suggestion
  bookmarks = {
    persist = false,
  },
```

Rather than a toggle and path, we can use just one option that is boolean or string. This is consistent with other options.

See [proposed help](https://github.com/nvim-tree/nvim-tree.lua/pull/3033#issuecomment-2661676946)

We then add the type explitictly so that it may be validated. Add to `ACCEPTED_TYPES`:
```lua
  bookmarks = {
    persist = { "boolean", "string" },
  },
```